subsys/net/lib/openthread/platform/radio.c

```diff
@@ -1019,23 +1019,113 @@ uint8_t otPlatRadioGetCslAccuracy(otInstance *aInstance)
+		0x00; ///< Thread Vendor-specific ACK Probing IE subtype ID
+	const uint32_t IE_VENDOR_THREAD_OUI = 0xeab89b; ///< Thread Vendor-specific IE OUI
+	const uint8_t IE_VENDOR_THREAD_RSSI_TOKEN =
+		0x01; ///< Thread Vendor-specific ACK Probing IE RSSI value placeholder
+	const uint8_t IE_VENDOR_THREAD_MARGIN_TOKEN =
+		0x02; ///< Thread Vendor-specific ACK Probing IE Link margin value placeholder
+	const uint8_t IE_VENDOR_THREAD_LQI_TOKEN =
+		0x03; ///< Thread Vendor-specific ACK Probing IE LQI value placeholder
+	const uint8_t ID_OFFSET = 7;
+	const uint8_t TYPE_OFFSET = 7;
+	const uint16_t ID_MASK = 0x00ff << ID_OFFSET;
+	const uint8_t LENGTH_MASK = 0x7f;
+	const uint8_t TYPE_MASK = 0x01 << TYPE_OFFSET;
+	const uint8_t HEADER_LEN = 2;
+	const uint8_t OUI_SIZE = 3;
+	const uint8_t SUB_TYPE = 1;
+	const uint8_t TYPE = 0x00;
+	uint8_t content_len;
+	uint16_t element_id = 0x0000;
+	uint8_t link_metrics_idx = 6;
+	uint8_t link_metric_data_len = (uint8_t)lqi + (uint8_t)link_margin + (uint8_t)rssi;
+
+	__ASSERT(link_metric_data_len <= 2, "Thread limits to 2 metrics at most");
+
+	if (link_metric_data_len == 0) {
+		return 0;
+	}
+
+	/* Set Element ID */
+	element_id = (((uint16_t)IE_VENDOR_ID) << ID_OFFSET) & ID_MASK;
+	sys_put_le16(element_id, &ie_header[0]);
+
+	/* Set Length - number of octets in content field. */
+	content_len = OUI_SIZE + SUB_TYPE + link_metric_data_len;
+	ie_header[0] = (ie_header[0] & ~LENGTH_MASK) | (content_len & LENGTH_MASK);
+
+	/* Set Type */
+	ie_header[1] = (ie_header[1] & ~TYPE_MASK) | (TYPE & TYPE_MASK);
+
+	/* Set Vendor Oui */
+	sys_put_le24((uint32_t)IE_VENDOR_THREAD_OUI, &ie_header[2]);
+
+	/* Set SubType */
+	ie_header[5] = IE_VENDOR_THREAD_ACK_PROBING_ID;
+
+	/* Set Link Metrics Tokens */
+	if (lqi) {
+		ie_header[link_metrics_idx++] = IE_VENDOR_THREAD_LQI_TOKEN;
+	}
+
+	if (link_margin) {
+		ie_header[link_metrics_idx++] = IE_VENDOR_THREAD_MARGIN_TOKEN;
+	}
+
+	if (rssi) {
+		ie_header[link_metrics_idx++] = IE_VENDOR_THREAD_RSSI_TOKEN;
+	}
+
+	return HEADER_LEN + content_len;
+}
+
 otError otPlatRadioConfigureEnhAckProbing(otInstance *aInstance, otLinkMetrics aLinkMetrics,
 					  const otShortAddress aShortAddress,
 					  const otExtAddress *aExtAddress)
 {
 	int result;
+	uint16_t ie_header_len;
+	uint8_t ie_header[OT_ACK_IE_MAX_SIZE];
+	uint8_t ext_addr_le[OT_EXT_ADDRESS_SIZE];
+	uint8_t short_addr_le[SHORT_ADDRESS_SIZE];
+	struct ieee802154_config config;
 
 	ARG_UNUSED(aInstance);
 
-	struct ieee802154_config config = {
-		.enh_ack.lqi = aLinkMetrics.mLqi,
-		.enh_ack.link_margin = aLinkMetrics.mLinkMargin,
-		.enh_ack.rssi = aLinkMetrics.mRssi,
-		.enh_ack.short_addr = aShortAddress,
-		.enh_ack.ext_addr = aExtAddress->m8,
-	};
+	ie_header_len = set_vendor_ie_header(aLinkMetrics.mLqi, aLinkMetrics.mLinkMargin,
+					     aLinkMetrics.mRssi, ie_header);
+
+	sys_put_le16(aShortAddress, short_addr_le);
+	sys_memcpy_swap(ext_addr_le, aExtAddress, OT_EXT_ADDRESS_SIZE);
```

Comment:
`otExtAddress` is a byte array, how to interpret endiannes in such case. How do we know we need to swap?